pkg/db/canary.go

```diff
@@ -139,11 +139,9 @@ func DeleteChecks(id []string) error {
 func GetCanary(id string) (*pkg.Canary, error) {
 	var model *pkg.Canary
 	if err := Gorm.Where("id = ?", id).First(&model).Error; err != nil {
-		if errors.Is(err, gorm.ErrRecordNotFound) {
```

Comment:
Why do we remove this - callers of this method should not be aware of the implementation ? I also think we can switch to Find which never throws a record not  found

Reply:
In general, FindByID db methods should either return the requested resource or return an error. A nil error means the resource was found. 

The previous implementation can potentially return 2 nil values which, in my opinion, is a bad practice as that can cause accidental panics. This requires 2 nil checks for all the callers.

I agree that callers needing to check against the gorm error is not a good practice. Ideally, we'd need a domain level "not found" error and the db methods would map `gorm.ErrRecordNotFound` error to our native NotFound error. But, we currently don't have such domain errors.

What do you think @moshloop

Reply:
I usually prefer:
Find -> can or cannot exist, not found should not return err
Get -> has to exist, not found raises an error

For fetch,some argue it should work like find and others argue it should work like get, I'd prefer we only use Find and Get  not use Fetch

Ref: https://dev.to/maikomiyazaki/beginner-s-cheat-sheet-commonly-used-verbs-for-naming-functions-methods-and-variables-509i

Reply:
@yashmehrotra That's a new one to me. Personally I would prefer to not have to deal with nuances like that. `Find, Get & Fetch` are I think used pretty much interchangeably. And, a method that conveys that the resource was not found either through errors or nil value should be sufficient to deal with any use cases.

@moshloop I guess both approaches inform the user that the resource was not found in different ways but I still think a `not found` error is more suitable simply to prevent accidentally accessing nil values after err check.

Reply:
> Find, Get & Fetch are I think used pretty much interchangeably
Have to disagree here, Find and Get are fundamentally different on a semantic grammar level, you can "Find" something but you may or may not have what you were looking for. When you "Get" something, the implication is that it already exists

Reply:
I'm not denying that there's a semantic difference between them. I'm just saying those are not standard verbs and are used synonymously in practice - even in our codebase.

Anyway, don't want to take too much time on this one. I have made changes.